src/language-markdown/print-preprocess.js

```diff
@@ -31,7 +31,7 @@ function transformInlineCode(ast, options) {
       return node;
     }
 
-    return { ...node, value: node.value.replace(/\s+/g, " ") };
+    return { ...node, value: node.value.replace(/\n/g, " ") };
```

Comment:
Maybe we can move this to `print`? So we can reduce one `mapAst` call. Can be separate PR.